quic/s2n-quic-transport/src/path/mod.rs

```diff
@@ -435,27 +435,20 @@ impl<Config: endpoint::Config> Path<Config> {
     //# A PL MUST NOT send a datagram (other than a probe
     //# packet) with a size at the PL that is larger than the current
     //# PLPMTU.
+
+    /// Clamps payload sizes to the current MTU for the path
+    ///
+    /// # Panics
+    ///
+    /// Panics if this is called when the path is amplification limited
     #[inline]
     pub fn clamp_mtu(&self, requested_size: usize, transmission_mode: transmission::Mode) -> usize {
-        let mtu = self.mtu(transmission_mode);
+        debug_assert!(
+            !self.at_amplification_limit(),
+            "amplification limits should be checked before clamping MTU values"
+        );
```

Comment:
but it only panics in debug mode right? so do we still need to Panic comment?

Reply:
I think it's worth documenting so you know when you can call it.